pkg/kine/drivers/generic/admission_test.go

```diff
@@ -0,0 +1,47 @@
+package generic
+
+import (
+	"context"
+	"testing"
+
+	"github.com/onsi/gomega"
+)
+
+func TestAllowAllPolicy_Admit(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	policy := &AllowAllPolicy{}
+
+	callOnFinish, err := policy.Admit(context.Background())
+
+	g.Expect(err).To(gomega.BeNil())
+	g.Expect(callOnFinish).ToNot(gomega.BeNil())
+}
+
+func TestLimitPolicy_Admit(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	t.Run("Admit when below limit", func(t *testing.T) {
+		policy := NewLimitPolicy(2)
+
+		callOnFinish, err := policy.Admit(context.Background())
+
+		g.Expect(err).To(gomega.BeNil())
+		g.Expect(callOnFinish).ToNot(gomega.BeNil())
+		g.Expect(policy.metrics.NumCurrentTxn.Load()).To(gomega.Equal(int64(1)))
+		callOnFinish()
+		g.Expect(policy.metrics.NumCurrentTxn.Load()).To(gomega.Equal(int64(0)))
+	})
+
+	t.Run("Deny when at limit", func(t *testing.T) {
+		policy := NewLimitPolicy(2)
+		policy.semaphore.Acquire(context.Background(), 2)
+		defer policy.semaphore.Release(2)
+		policy.metrics.NumCurrentTxn.Store(2)
+
+		callOnFinish, err := policy.Admit(context.Background())
+
+		g.Expect(err).To(gomega.HaveOccurred())
+		g.Expect(callOnFinish).ToNot(gomega.BeNil())
+		g.Expect(policy.metrics.NumCurrentTxn.Load()).To(gomega.Equal(int64(2)))
+	})
```

Comment:
This seems to be very closely tied to the implementation itself. How about a test where we test the functionality instead?

e.g: on limit 2, first two succeed, third one fails, one completes then a new one can succeed?